ethcore/trace/src/types/trace.rs

```diff
@@ -51,6 +103,49 @@ impl CreateResult {
 	}
 }
 
+/// `Create` method. Distinguish between use of `CREATE` and `CREATE2` opcodes in an action.
+#[derive(Debug, Clone, PartialEq)]
+pub enum CreationMethod {
+	/// Create
+	Create,
+	/// Create2
+	Create2,
+}
+
+impl TryFrom<ActionType> for CreationMethod {
+	type Error = &'static str;
+	fn try_from(action_type: ActionType) -> Result<Self, Self::Error> {
+		match action_type {
+			ActionType::Call => Err("Call cannot be converted to CreationMethod"),
+			ActionType::CallCode => Err("CallCode cannot be converted to CreationMethod"),
+			ActionType::DelegateCall => Err("DelegateCall cannot be converted to CreationMethod"),
+			ActionType::StaticCall => Err("StaticCall cannot be converted to CreationMethod"),
+			ActionType::Create => Ok(CreationMethod::Create),
+			ActionType::Create2 => Ok(CreationMethod::Create2),
+		}
+	}
+}
+
+impl Encodable for CreationMethod {
+	fn rlp_append(&self, s: &mut RlpStream) {
+		let v = match *self {
+			CreationMethod::Create => 0u32,
+			CreationMethod::Create2 => 1,
+		};
+		Encodable::rlp_append(&v, s);
+	}
+}
+
+impl Decodable for CreationMethod {
+	fn decode(rlp: &Rlp) -> Result<Self, DecoderError> {
+		rlp.as_val().and_then(|v| Ok(match v {
+			0u32 => CreationMethod::Create,
+			1 => CreationMethod::Create2,
+			_ => return Err(DecoderError::Custom("Invalid value of RewardType item")),
+		}))
+	}
+}
+
 /// Description of a _call_ action, either a `CALL` operation or a message transaction.
 #[derive(Debug, Clone, PartialEq, RlpEncodable, RlpDecodable)]
 pub struct Call {
```

Comment:
What do you think about annotating that data of this type end up rlp-ized in the database and thus any changes need to be made backwards compatibly?